src/games/behavmixed.imp

```diff
@@ -643,130 +589,119 @@ T MixedBehaviorProfile<T>::DiffNodeValue(const GameNode &p_node,
+      ComputePass1_realizProbs(childNode);
+    }
+  }
+}
+
 template <class T>
-void MixedBehaviorProfile<T>::ComputeSolutionDataPass2(const GameNode &node) const
+void MixedBehaviorProfile<T>::ComputePass2_beliefs_nodeValues_actionValues(const GameNode &node) const
 {
   if (node->GetOutcome()) {
     GameOutcome outcome = node->GetOutcome();
-    for (int pl = 1; pl <= m_support.GetGame()->NumPlayers(); pl++) {
-      m_nodeValues(node->GetNumber(), pl) += static_cast<T>(outcome->GetPayoff(pl));
+    for (auto player: m_support.GetGame()->GetPlayers()) {
+      map_nodeValues[node][player] += static_cast<T>(outcome->GetPayoff(player));
     }
   }
 
   GameInfoset iset = node->GetInfoset();
 
   if (iset) {
     T infosetProb = (T) 0;
-    for (int i = 1; i <= iset->NumMembers(); i++) {
-      infosetProb += m_realizProbs[iset->GetMember(i)->GetNumber()];
+    for (auto i: iset->GetMembers()) {
+      infosetProb += map_realizProbs[i];
     }
 
-    if (infosetProb != infosetProb * (T) 0) {
-      m_beliefs[node->GetNumber()] = m_realizProbs[node->GetNumber()] / infosetProb;
+    //if (infosetProb != infosetProb * (T) 0) {
+    if (infosetProb != (T) 0) {
+      map_beliefs[node] = map_realizProbs[node] / infosetProb;
     }
 
     // push down payoffs from outcomes attached to non-terminal nodes
-    for (int child = 1; child <= node->NumChildren(); child++) {
-      m_nodeValues.SetRow(node->GetChild(child)->GetNumber(),
-			  m_nodeValues.Row(node->GetNumber()));
+    for (auto child: node->GetChildren()) {
+      map_nodeValues[child] = map_nodeValues[node];
     }
 
-    for (int pl = 1; pl <= m_support.GetGame()->NumPlayers(); pl++) {
-      m_nodeValues(node->GetNumber(), pl) = (T) 0;
+    for (auto player: m_support.GetGame()->GetPlayers()) {
+      map_nodeValues[node][player] = (T) 0;
     }
 
-    for (int child = 1; child <= node->NumChildren(); child++)  {
-      GameNode childNode = node->GetChild(child);
-      ComputeSolutionDataPass2(childNode);
+    for (auto child: node->GetChildren())  {
+      ComputePass2_beliefs_nodeValues_actionValues(child);
 
-      GameAction act = childNode->GetPriorAction();
+      GameAction act = child->GetPriorAction();
 
-      for (int pl = 1; pl <= m_support.GetGame()->NumPlayers(); pl++) {
-        m_nodeValues(node->GetNumber(), pl) +=
-          GetActionProb(act) * m_nodeValues(childNode->GetNumber(), pl);
+      for (auto player: m_support.GetGame()->GetPlayers()) {
+        map_nodeValues[node][player] += GetActionProb(act) * map_nodeValues[child][player];
       }
 
       if (!iset->IsChanceInfoset()) {
-	T &cpay = m_actionValues(act->GetInfoset()->GetPlayer()->GetNumber(),
-				 act->GetInfoset()->GetNumber(),
-				 act->GetNumber());
-	if (infosetProb != infosetProb * (T) 0) {
-	  cpay += m_beliefs[node->GetNumber()] * m_nodeValues(childNode->GetNumber(), iset->GetPlayer()->GetNumber());
-	}
-	else {
-	  cpay = (T) 0;
-	}
+        if (infosetProb != infosetProb * (T) 0) {
```

Comment:
Can use a conditional ternary expression here.  Also can get rid of the multiply-by-zero indirection (as done elsewhere)